src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,111 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size;
+    private T[] dataArray;
+
+    public ArrayList() {
+        dataArray = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == dataArray.length) {
+            extensionArray();
+        }
+        dataArray[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexInclusive(index);
+        if (size == dataArray.length) {
+            extensionArray();
+        }
+
+        if (index != size) {
+            System.arraycopy(dataArray, index, dataArray, index + 1, size - index);
+        }
```

Comment:
may be simplified and method checkIndexInclusive() is not really required at all:
```suggestion
        if (size == dataArray.length) {
            add(value);
            return;
        }
        checkIndex(index);
        System.arraycopy(dataArray, index, dataArray, index + 1, size - index);
```